src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -318,14 +320,18 @@ def _get_csv_dmatrix_pipe_mode(pipe_path, csv_weights):
             examples = []
             for example in reader:
                 # Write each feature (column) of example into a single numpy array
-                tmp = [as_numpy(feature).squeeze() for feature in example]
-                tmp = np.array(tmp)
-                if len(tmp.shape) > 1:
-                    # Columns are written as rows, needs to be transposed
-                    tmp = tmp.T
-                else:
+                tmp = np.column_stack([as_numpy(f) for f in example])
+
+                # If a subsample ratio is specified, only take the first
+                # `BATCH_SIZE * subsample_on_read` records and discard the rest
```

Comment:
I thought the intention was to perform some systematic sampling (i.e. take every nth row) instead of taking the first group of records.

Reply:
After having further discussions this approach found to be more flexible.